src/share/classes/com/sun/javatest/regtest/agent/JUnitRunner.java

```diff
@@ -91,4 +109,40 @@ public static void main(ClassLoader loader, String... args) throws Exception {
         }
     }
 
+    private static void runWithJUnitPlatform(Class<?> mainClass) throws Exception {
+        // https://junit.org/junit5/docs/current/user-guide/#launcher-api-execution
+        Thread.currentThread().setContextClassLoader(mainClass.getClassLoader());
```

Comment:
this line, to set the context class loader, might be an issue for some core-libs tests.

Reply:
... I guess it might just mean we can't use JUnit Jupiter for some of the low-level core-libs tests

Reply:
Yes, either that or we find another way to make those core-libs tests and JUnit happy by agreeing upon a class loader.

Reply:
If there is a way to avoid using the context class loader and agree on some other one, that might be worth investigating as a followup.